brainbuilder/utils/sonata/reindex.py

```diff
@@ -496,3 +498,119 @@ def write_section_types(morphologies, population, direction, edge_path):
         pop0 = h5[f"edges/{population}/0"]
         field = direction + "_section_type"
         backup_and_create_dataset(pop0, field, types, np.uint32)
+
+
+def _compute_center_point(morph, section_ids, segment_ids, segment_offsets):
+    """Compute center point along the segment."""
+    segment_start, segment_end = np.zeros((2, len(section_ids), 3))
+
+    for i, (section_id, segment_id) in enumerate(zip(section_ids, segment_ids)):
+        if section_id == 0:
+            segment_start[i] = morph.soma.center
+            segment_end[i] = segment_start[i] + 1
+        else:
+            section = morph.section(section_id - 1)  # off-by-one gotcha
+
+            # if last segment
+            if segment_id == len(section.points) - 1:
+                segment_id -= 1
```

Comment:
why does the last `segment_id` need to be decremented? can you add a comment?

Reply:
Good catch, I had messed it up myself, actually. It is fixed and commented now.